src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -112,11 +117,54 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             );
         }
         try {
-            XContentParser parser = request.contentParser();
-            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
-            Template template = Template.parse(parser);
 
-            WorkflowRequest workflowRequest = new WorkflowRequest(workflowId, template, validation, provision, params);
+            Template template;
+            Map<String, String> useCaseDefaultsMap = Collections.emptyMap();
+            if (useCase != null) {
+                String json = ParseUtils.resourceToString("/" + DefaultUseCases.getSubstitutionReadyFileByUseCaseName(useCase));
+                String defaultsFilePath = DefaultUseCases.getDefaultsFileByUseCaseName(useCase);
+                useCaseDefaultsMap = ParseUtils.parseJsonFileToStringToStringMap("/" + defaultsFilePath);
+
+                if (request.hasContent()) {
+                    try {
+                        XContentParser parser = request.contentParser();
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        Map<String, String> userDefaults = ParseUtils.parseStringToStringMap(parser);
+                        // updates the default params with anything user has given that matches
+                        for (Map.Entry<String, String> userDefaultsEntry : userDefaults.entrySet()) {
+                            if (useCaseDefaultsMap.containsKey(userDefaultsEntry.getKey())) {
+                                useCaseDefaultsMap.put(userDefaultsEntry.getKey(), userDefaultsEntry.getValue());
+                            }
+                        }
+                    } catch (Exception ex) {
+                        String errorMessage = "failure parsing request body when a use case is given";
+                        logger.error(errorMessage, ex);
+                        throw new FlowFrameworkException(errorMessage, ExceptionsHelper.status(ex));
```

Comment:
This may throw an internal server error if the parseStringToStringMap throws an IOException. The ExceptionsHelper has no case for IOException and defaults the status code to 500. Lets change that

Reply:
changes so rest status is BAD_REQUEST when IO exception